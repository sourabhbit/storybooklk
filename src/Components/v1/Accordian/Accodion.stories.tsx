import { ComponentMeta, ComponentStory } from '@storybook/react'
import { generateArgTypes } from 'Helpers/Storybook'
import Accordions from './index'
import { filters } from './mockValue'

export default {
	title: 'Components/v1/Accodion',
	component: Accordions,
	argTypes: {
		id: generateArgTypes({
			type: 'string',
			required: true,
			description: 'String used as unique identifier',
			category: 'General',
		}),
	},
} as ComponentMeta<typeof Accordions>

const Template: ComponentStory<typeof Accordions> = (args) => (
	<Accordions {...args} />
)
export const Overview = Template.bind({})
Overview.args = {
	id: 'AccodionGroup',
	filters: filters,
}
