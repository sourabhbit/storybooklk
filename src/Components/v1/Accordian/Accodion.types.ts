import { ReactNode } from 'react'

export interface AccodionGroupTypes {
	id: string
	filters: Array<FilterTypes>
}

export interface FilterTypes {
	name: string
	options: Array<OptionTypes>
}
export interface OptionTypes {
	title: string
	no_of_products: number
}

export interface AccodionTypes {
	id: string
	children: ReactNode
	title: string
	expand: boolean
	handleExpand: () => void
}
