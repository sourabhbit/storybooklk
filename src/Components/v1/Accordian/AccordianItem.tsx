import { AccodionTypes } from './Accodion.types'
import {
	AccordionnWrapper,
	TitleBox,
	TitleWrapper,
	Title,
	Content,
	DownArrow,
	UpArrow,
} from './styles'

const Accordion = ({
	id,
	children,
	title,
	expand,
	handleExpand,
}: AccodionTypes) => {
	return (
		<AccordionnWrapper id={id}>
			<TitleBox onClick={() => handleExpand()} expand={expand}>
				<TitleWrapper>
					<Title>{title}</Title>
				</TitleWrapper>
				<i>{!expand ? <UpArrow /> : <DownArrow />}</i>
			</TitleBox>
			{expand && <Content>{children}</Content>}
		</AccordionnWrapper>
	)
}

export default Accordion
