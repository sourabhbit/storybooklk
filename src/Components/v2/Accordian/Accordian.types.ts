import { ComponentSizeENUM, ThemeENUM, TypographyENUM } from 'Types/general'
export interface AccordianTypes {
	id: string
	text: string
	icon?: string
	options: string[]
	theme: ThemeENUM
	font: TypographyENUM
	componentSize?: ComponentSizeENUM
}
