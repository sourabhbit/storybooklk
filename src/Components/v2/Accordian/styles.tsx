import styled from 'styled-components'
import { ComponentSizeENUM, ThemeENUM, TypographyENUM } from 'Types/general'

const getFontSize = (size: ComponentSizeENUM) => {
	if (size === ComponentSizeENUM.small) {
		return 'var(--fs-12)'
	} else if (size === ComponentSizeENUM.medium) {
		return 'var(--fs-18)'
	} else {
		return 'var(--fs-30)'
	}
}

export const AccordianContainer = styled.div<{
	styleColor: ThemeENUM
	styleSize: ComponentSizeENUM
	styleFont: TypographyENUM
}>`
	display: flex;
	flex-direction: column;
	font-size: ${(props) => getFontSize(props.styleSize)};
	font-family: ${(props) => props.styleFont};
	height: 100px;
`

export const AccordianHeader = styled.div`
	display: flex;
	align-items: center;
	justify-content: space-between;
	border: none;
	outline: none;
	text-align: center;
	cursor: pointer;
`

export const AccordianText = styled.span<{
	// styleColor: ThemeENUM
	styleSize: ComponentSizeENUM
	// styleFont: TypographyENUM
}>`
	font-size: ${(props) => getFontSize(props.styleSize)};
`

export const AccordianIcon = styled.span`
	width: 36px;
	height: 24px;
	border: 1px solid var(--dark-blue-100);
	border-radius: 207.782px;
	display: flex;
	align-items: center;
	justify-content: center;
`

export const OptionWrapper = styled.span`
	display: flex;
	flex-direction: column;
	padding-top: var(--pd-20);
`

export const Option = styled.span<{
	styleSize: ComponentSizeENUM
	styleColor: ThemeENUM
}>`
	height: 40px;
	font-size: ${(props) => getFontSize(props.styleSize)};
`
