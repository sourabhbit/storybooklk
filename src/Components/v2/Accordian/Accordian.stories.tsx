import { ComponentMeta, ComponentStory } from '@storybook/react'
import { generateArgTypes, generateGeneralArgs } from 'Helpers/Storybook'
import { ComponentSizeENUM, ThemeENUM, TypographyENUM } from 'Types/general'
import Accordian from './index'

export default {
	title: 'Components/v2/Accordian',
	component: Accordian,
	argTypes: {
		...generateGeneralArgs(),
		text: generateArgTypes({
			type: 'string',
			required: true,
			description: 'Show accordian heading',
			category: 'Component',
		}),
		icon: generateArgTypes({
			type: 'string',
			required: false,
			description: 'Show icon',
			category: 'Component',
		}),
		options: generateArgTypes({
			type: 'array',
			required: true,
			description: 'Show accordian options',
			category: 'Component',
		}),
	},
} as ComponentMeta<typeof Accordian>

const Template: ComponentStory<typeof Accordian> = (arges) => (
	<Accordian {...arges} />
)

export const Overview = Template.bind({})
Overview.args = {
	id: 'accordian-id',
	text: 'Included',
	componentSize: ComponentSizeENUM.small,
	theme: ThemeENUM.secondary,
	font: TypographyENUM.serif,
	options: [
		'Free shipping & returns',
		'365 day warranty',
		'Money back gurantee',
	],
}
