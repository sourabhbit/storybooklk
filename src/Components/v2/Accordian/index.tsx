import {
	AccordianContainer,
	AccordianHeader,
	AccordianText,
	AccordianIcon,
	Option,
	OptionWrapper,
} from './styles'
import { AccordianTypes } from './Accordian.types'
import { ComponentSizeENUM, ThemeENUM, TypographyENUM } from 'Types/general'
import Up from '../../../../icons/upArrow.svg'
import Down from '../../../../icons/downArrow.svg'
import React from 'react'

const Accordian = ({
	id,
	options,
	text,
	icon,
	componentSize = ComponentSizeENUM.small,
	theme = ThemeENUM.primary,
	font = TypographyENUM.serif,
}: AccordianTypes) => {
	const [showOptions, setShowOptions] = React.useState(false)

	const onChangeaction = () => {
		setShowOptions(!showOptions)
	}

	return (
		<AccordianContainer
			id={id}
			data-testid="accordian-container"
			styleColor={theme}
			styleFont={font}
			styleSize={componentSize}
		>
			<AccordianHeader onClick={() => onChangeaction()}>
				<AccordianText styleSize={componentSize}>{text}</AccordianText>
				<AccordianIcon>
					<img alt="key" src={showOptions ? Up : Down} />
				</AccordianIcon>
			</AccordianHeader>
			<OptionWrapper>
				{showOptions &&
					options?.map((option, index) => (
						<div data-testid="option-wrapper" key={index}>
							{icon && <img alt="bullet" src={icon} />}
							<Option styleSize={componentSize} styleColor={theme}>
								{option}
							</Option>
						</div>
					))}
			</OptionWrapper>
		</AccordianContainer>
	)
}

export default Accordian
