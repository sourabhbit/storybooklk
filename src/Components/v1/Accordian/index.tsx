import { useState } from 'react'
import { AccodionGroupTypes, FilterTypes, OptionTypes } from './Accodion.types'
import Accordion from './AccordianItem'
import { FlexCol, Flex, CheckBoxLabel, Label } from './styles'

const Accordions = ({ id, filters }: AccodionGroupTypes) => {
	const [expandIndex, setExpandIndex] = useState(0)
	const handleExpand = (index: number) => {
		console.log(index, expandIndex)
		if (index === expandIndex) {
			setExpandIndex(0)
		} else setExpandIndex(index)
	}

	return (
		<>
			{filters.map((item: FilterTypes, index: number) => {
				return (
					<Accordion
						title={item.name}
						expand={expandIndex === index + 1}
						handleExpand={() => handleExpand(index + 1)}
						key={index + 1}
						id={id}
					>
						<FlexCol>
							{item.options.map((item: OptionTypes, index: number) => {
								return (
									<Flex key={index}>
										<Label>
											<input
												type="checkbox"
												name={item.title}
												id=""
											/>
											<CheckBoxLabel>
												{item.title +
													'(' +
													item.no_of_products +
													')'}
											</CheckBoxLabel>
										</Label>
									</Flex>
								)
							})}
						</FlexCol>
					</Accordion>
				)
			})}
		</>
	)
}

export default Accordions
