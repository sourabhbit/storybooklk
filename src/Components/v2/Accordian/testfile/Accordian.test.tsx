import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import { ComponentSizeENUM, ThemeENUM, TypographyENUM } from 'Types/general'
import Accordian from '../index'

const props = {
	id: 'accordian-id',
	text: 'Included',
	componentSize: ComponentSizeENUM.small,
	theme: ThemeENUM.secondary,
	font: TypographyENUM.serif,
	options: [
		'Free shipping & returns',
		'365 day warranty',
		'Money back gurantee',
	],
}

describe('<Accordian />', () => {
	let _asFragment: any = null

	beforeEach(() => {
		const { asFragment } = render(<Accordian {...props} />)
		_asFragment = asFragment
	})

	afterAll(() => {
		_asFragment = null
	})

	it('Should render component', () => {
		const container = screen.getByTestId('accordian-container')
		// const option = screen.queryByTestId('option-wrapper')
		expect(container).toBeInTheDocument()
		// expect(option).toBeInTheDocument()
	})

	it('Should take a snapshot', () => {
		expect(_asFragment()).toMatchSnapshot()
	})
})
