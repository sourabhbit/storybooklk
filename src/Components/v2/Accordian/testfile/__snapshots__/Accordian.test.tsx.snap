// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordian /> Should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL cKDkGq"
    data-testid="accordian-container"
    id="accordian-id"
  >
    <div
      class="sc-gsDKAQ brBwKm"
    >
      <span
        class="sc-dkPtRN fobWJP"
      >
        Included
      </span>
      <span
        class="sc-hKwDye feUFAa"
      >
        <img
          alt="key"
        />
      </span>
    </div>
    <span
      class="sc-eCImPb cFVUxq"
    />
  </div>
</DocumentFragment>
`;
